src/rs.cpp

```diff
@@ -1204,6 +1216,56 @@ const char* rs2_get_option_value_description(const rs2_options* options, rs2_opt
 }
 HANDLE_EXCEPTIONS_AND_RETURN(nullptr, options, option, value)
 
+rs2_subscription * rs2_set_options_changed_callback( const rs2_options * options,
+                                                     rs2_option_value_update_callback_ptr callback,
+                                                     rs2_error ** error ) BEGIN_API_CALL
+{
+    VALIDATE_NOT_NULL( options );
+    VALIDATE_NOT_NULL( callback );
+    auto sens = dynamic_cast< synthetic_sensor * >( options->options );
```

Comment:
Shouldn't have a dynamic cast to synthetic here...

Reply:
Using `sensor_interface`